src/Timers.cpp

```diff
@@ -606,6 +607,29 @@ PVR_ERROR Timers::AddTimer(const kodi::addon::PVRTimer& timer)
     marginEnd = m_settings.m_defaultPostPadding;
   }
 
+  time_t endTime = timer.GetEndTime();
+
+  if (timer.GetEndTime() - timer.GetStartTime() <= 0 && timer.GetTimerType() == TIMER_ONCE_MANUAL)
+  {
+    std::vector<std::string> options;
+    int offset = 0;
+    for (unsigned int minutes = 30; minutes <= 180; minutes += 30)
+    {
+      const std::string option = kodi::tools::StringUtils::Format("%d %s", minutes, kodi::GetLocalizedString(12391).c_str());
+      options.emplace_back(option);
+    }
+    int option = kodi::gui::dialogs::Select::Show(kodi::GetLocalizedString(30201), options, 1);
+    if (option > 0)
```

Comment:
Should this not be >= 0?

-1 is returned if nothing is selected.

Reply:
Good catch @phunkyfish thanks.  I was going back and fourth between different options, number entry, time entry, so missed this.  It is too bad there is no slider dialog in the GUI.